sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ViewDescriptorFactory.scala

```diff
@@ -90,21 +90,36 @@ private[impl] object ViewDescriptorFactory extends ComponentDescriptorFactory {
           .map { method =>
             // validate that all updates return the same type
             val valueEntityClass = method.getAnnotation(classOf[Subscribe.ValueEntity]).value().asInstanceOf[Class[_]]
+            val valueEntityEventClass = valueEntityClass.getGenericSuperclass
+              .asInstanceOf[ParameterizedType]
+              .getActualTypeArguments
+              .head
+              .asInstanceOf[Class[_]]
             previousValueEntityClass match {
               case Some(`valueEntityClass`) => // ok
               case Some(other) =>
                 throw InvalidComponentException(
                   s"All update methods must return the same type, but [${method.getName}] returns [${valueEntityClass.getName}] while a previous update method returns [${other.getName}]")
               case None => previousValueEntityClass = Some(valueEntityClass)
             }
-            // FIXME validate that transform method accepts value entity state type
+
+            method.getParameterTypes.toList match {
+              case p1 :: Nil if p1 != valueEntityEventClass =>
+                throw InvalidComponentException(
```

Comment:
maybe not worthy but as they are the same error message would be beneficial to have a val with this exception. The idea behind this would be for the one coming next looking at this code not to have to read the whole exception to see if they have anything different. 
Feel free to skip this comment.

Reply:
Not a `val`, but a `def`.

A `val` will create the exception eagerly.

Reply:
I was in doubt about having 2 different msgs or not.. I will do this on a follow up PR.. just not to drag this one more.